### ISSUE TITLE
Some lineBreak rules related to ClassDeclaration MethodDefinition ignored

### ISSUE THREAD

**Issue by MaybiusStrip** [millermedeiros/esformatter#417]
Many rules related to `ClassDeclaration` and `MethodDefinition` seem to be ignored. I discovered this while trying to automate this transformation:
```diff
class Awesome {
  
  awesomeMethod() {
    // things
  }
+
}
```

Some of the things I tried include:
```
lineBreak.before.ClassDeclarationClosing = 2
lineBreak.before.ClassDeclarationClosingBrace = 2
lineBreak.after.MethodDefinition = 2
lineBreak.after.MethodDefinitionClosing = 2
lineBreak.after.MethodDefinitionClosingBrace = 2
```

These are working fine (but solve unrelated problems):
```
lineBreak.before.ClassDeclaration = 2
lineBreak.before.MethodDefinition = 2
whiteSpace.after.MethodDefinition = 2 // Can test this by adding a semicolon after the MethodDefinition
```

**Comment by millermedeiros** [comment]
**TL;DR;** the order that things are processed affects the outcome drastically, many of the supported options conflicts with each other... we should probably add support for `MethodDefinitionClosingBrace`

also, `ClassDeclarationClosingBrace` should actually be `ClassClosingBrace` since we use the same [setting for `ClassDeclaration` and `ClassExpression`](https://github.com/millermedeiros/esformatter/blob/master/lib/hooks/ClassDeclarationAndExpression.js#L22)

---

## overview of how transform works and conflicts:

the transform starts from the *leaf nodes* and moves down the tree - that gives us the option to undo any undesired line break/space inserted by the child elements, this is important since we actually need to add/remove line breaks before inserting any white spaces or indentation; by starting from leaf nodes we can do line breaks and white spaces in the same pass...

we also [add the `lineBreak.[before|after].MethodDefinition` before](https://github.com/millermedeiros/esformatter/blob/master/lib/transform.js#L61) calling the [`hook` for `MethodDefinition`](https://github.com/millermedeiros/esformatter/blob/master/lib/hooks/MethodDefinition.js) - this is done on purpose, to allow the hook itself to undo any conflicting change - maybe we should also change this behavior, and do the automatic line breaks after the `hook.format` call... But the `MethodDefinition.value` in this case is actually a [FunctionExpression](https://github.com/millermedeiros/esformatter/blob/master/lib/hooks/FunctionExpression.js) - you would need to target the `FunctionExpressionClosingBrace` since we don't support `MethodDefinitionClosingBrace` (which we probably should)

**protip:** best way to find supported options is to open the [`hooks` files and look for the `limit(Before|After)?` calls](https://github.com/millermedeiros/esformatter/tree/master/lib/hooks), also use [astexplorer](https://astexplorer.net/)

**Comment by millermedeiros** [comment]
this will be doable on the next version of esformatter!

```json
{
  "lineBreak": {
    "before": {
      "MethodDefinition": 3
    },
    "after": {
      "MethodDefinitionClosingBrace": 3
    }
  }
}
```

please note that for some weird reason the `lineBreak.after.MethodDefinition` still doesn't work as expected for the last `MethodDefinition` in the class... so I removed it from the default settings and added the `lineBreak.after.MethodDefinitionClosingBrace = ">= 1"` instead.
